segmentstore/server/src/main/java/io/pravega/segmentstore/server/tables/TableEntryDeltaIterator.java

```diff
@@ -0,0 +1,188 @@
+        val entry = getNextEntryFromBatch();
+        if (entry != null) {
+            return CompletableFuture.completedFuture(entry);
+        }
+
+        return fetchNextTableEntriesBatch().thenApply(val -> getNextEntryFromBatch());
+    }
+
+    private synchronized Map.Entry<DeltaIteratorState, TableEntry> getNextEntryFromBatch() {
+        if (this.currentEntry != null) {
+            val next = this.currentEntry.next();
+            if (!this.currentEntry.hasNext()) {
+                this.currentEntry = null;
+            }
+            return next;
+        }
+
+        return null;
+    }
+
+    private CompletableFuture<Void> fetchNextTableEntriesBatch() {
+        return toEntries(currentBatchOffset)
+                .thenAccept(entries -> {
+                    if (!entries.isEmpty()) {
+                        this.currentEntry = entries.iterator();
+                    } else {
+                        this.currentEntry = null;
+                    }
+                });
+    }
+
+    private CompletableFuture<List<Map.Entry<DeltaIteratorState, TableEntry>>> toEntries(long startOffset) {
+        TimeoutTimer timer = new TimeoutTimer(this.fetchTimeout);
+        int length = Math.min(maxLength, MAX_READ_SIZE);
+
+        if (endOfSegment()) {
+            return CompletableFuture.completedFuture(Collections.emptyList());
+        }
+        ReadResult result = this.segment.read(startOffset, length, timer.getRemaining());
+        return AsyncReadResultProcessor.processAll(result, this.executor, timer.getRemaining())
+                .thenApply(data -> parseEntries(data, startOffset, length));
+    }
+
+    @SneakyThrows(IOException.class)
+    private List<Map.Entry<DeltaIteratorState, TableEntry>> parseEntries(BufferView data, long startOffset, int readLength) {
+
+        long currentOffset = startOffset;
+        final long maxOffset = startOffset + readLength;
+
+        BufferView.Reader input = data.getBufferViewReader();
+        List<Map.Entry<DeltaIteratorState, TableEntry>> entries = new ArrayList<>();
+        try {
+            while (currentOffset < maxOffset) {
+                val entry = AsyncTableEntryReader.readEntryComponents(input, currentOffset, this.entrySerializer);
+                boolean reachedEnd = currentOffset + entry.getHeader().getTotalLength() >= this.maxLength + startOffset;
+                // We must preserve deletions to accurately construct a delta.
+                BufferView value = entry.getValue() == null ? BufferView.empty() : entry.getValue();
+                currentOffset += entry.getHeader().getTotalLength();
+                entries.add(new AbstractMap.SimpleEntry<>(
+                        new DeltaIteratorState(currentOffset, reachedEnd, this.shouldClear, entry.getHeader().isDeletion()),
+                        TableEntry.versioned(entry.getKey(), value, entry.getVersion())));
+            }
+
+        } catch (BufferView.Reader.OutOfBoundsException ex) {
```

Comment:
I believe I know why you are silently ignoring this exception, but please add a comment explaining why.